omnigibson/systems/micro_particle_system.py

```diff
@@ -1589,6 +1629,76 @@ class Cloth(MicroParticleSystem):
     """
     Particle system class to simulate cloth.
     """
+    @classmethod
+    def clothify_mesh_prim(cls, mesh_prim, remesh=True, particle_distance=None):
+        """
+        Clothifies @mesh_prim by applying the appropriate Cloth API, optionally re-meshing the mesh so that the
+        resulting generated particles are roughly @particle_distance apart from each other.
+
+        Args:
+            mesh_prim (Usd.Prim): Mesh prim to clothify
+            remesh (bool): If True, will remesh the input mesh before converting it into a cloth
+            particle_distance (None or float): If set and @remesh is True, specifies the absolute target distance
+                between generated cloth particles. If None, a value is automatically chosen such that the generated
+                cloth particles are roughly touching each other, given cls.particle_contact_offset and
+                @mesh_prim's scale
+        """
+        # Possibly remesh if requested
+        if remesh:
+            # We will remesh in pymeshlab, but it doesn't allow programmatic construction of a mesh with texcoords so
+            # we convert our mesh into a trimesh mesh, then export it to a temp file, then load it into pymeshlab
+            tm = mesh_prim_to_trimesh_mesh(mesh_prim=mesh_prim, include_normals=True, include_texcoord=True)
+            # Tmp file written to: {tmp_dir}/{tmp_fname}/{tmp_fname}.obj
+            tmp_name = f"{mesh_prim.GetName()}_{datetime.datetime.now().strftime('%Y-%m-%d_%H-%M-%S')}"
+            tmp_dir = os.path.join(tempfile.gettempdir(), tmp_name)
+            tmp_fpath = os.path.join(tmp_dir, f"{tmp_name}.obj")
+            Path(tmp_dir).mkdir(parents=True, exist_ok=True)
+            tm.export(tmp_fpath)
+            ms = pymeshlab.MeshSet()
+            ms.load_new_mesh(tmp_fpath)
+
+            # Re-mesh based on @particle_distance - distance chosen such that at rest particles should be just touching
+            # each other. The 1.5 magic number comes from the particle cloth demo from omni
+            particle_distance = cls.particle_contact_offset * 2 / (1.5 * np.mean(mesh_prim.GetAttribute("xformOp:scale").Get())) \
+                if particle_distance is None else particle_distance
+            ms.meshing_isotropic_explicit_remeshing(iterations=20, targetlen=pymeshlab.AbsoluteValue(particle_distance))
```

Comment:
What does `iterations=20` do here? Can we keep iterating until targetlen is satisfied?

Reply:
assert targetlen is reached